### PR TITLE
haskellPackages.hledger_1_27_1: Fix dependency toward hledger-lib

### LINKED ISSUE #200000: hledger_1_27_1 doesn't build
### Steps To Reproduce
1. build hledger_1_27_1

### Build log

You get a failure because hledger depends on hledger_lib which should be in version 1_27_1. There is a hledger_lib_1_27_1 that hledger_1_27_1 should depend on.

### Notify maintainers

@sternenseemann @cdepillabout 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
querying info about '/nix/store/m142rmlj114xwgwgf6k2gvvm33zxkbia-nix-info' on 'https://cache.nixos.org'...
downloading 'https://cache.nixos.org/m142rmlj114xwgwgf6k2gvvm33zxkbia.narinfo'...
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/m142rmlj114xwgwgf6k2gvvm33zxkbia-nix-info
copying path '/nix/store/m142rmlj114xwgwgf6k2gvvm33zxkbia-nix-info' from 'https://cache.nixos.org'...
downloading 'https://cache.nixos.org/nar/1xiaxp2dk3zs0418sc8k1fmsy3lzk26xwmcsknb62145nrgj8dgn.nar.xz'...
 - system: `"x86_64-linux"`
 - host os: `Linux 6.0.5-200.fc36.x86_64, Fedora Linux, 36 (Workstation Edition), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(root): `""`
 - nixpkgs: `/home/cassou/Documents/projects/nix-system/nixpkgs`

```

### PR DESCRIPTION
**Pull request by DamienCassou**
###### Description of changes

Issue being fixed: https://github.com/NixOS/nixpkgs/issues/200000.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by DamienCassou** [comment]
Result of `nixpkgs-review pr 200055` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -1577,6 +1577,10 @@ self: super: {
     ghc-paths = lsuper.ghc-paths.override { Cabal = null; };
   });
 
+  hledger_1_27_1 = super.hledger_1_27_1.overrideScope  (lself: lsuper: {
```
Comment (sternenseemann) [inline pkgs/development/haskell-modules/configuration-common.nix:1580]: Can you wrap this in a `doDistribute` call, so it is built by Hydra? Additionally, I'm fairly sure a plain `.override` would be enough which would be preferrable.

Comment (DamienCassou) [reply pkgs/development/haskell-modules/configuration-common.nix]: fixed

**Comment by sternenseemann** [comment]
@ofborg build haskellPackages.hledger_1_27_1

**Comment by DamienCassou** [comment]
Thank you for your feedback @sternenseemann. I think I did what you recommended.

**Pull request closed without merging**